### PR TITLE
refactor: try using strategy design pattern

### PR DESCRIPTION
**Pull request by LiutkauskasMtelesoftas**

### CODE REVIEW

File: python/Interfaces/UpdateStrategyInterface.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from abc import ABC, abstractmethod
+
+class UpdateStrategyInterface:
+    def update(self, item):
```
Comment (vytenisgelazius) [inline python/Interfaces/UpdateStrategyInterface.py]: nit: return type hint

File: python/Strategies/backstage_passes.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import sys
+sys.path.append('Interfaces')
+
+from Interfaces import UpdateStrategyInterface
+
+class BackstagePassesUpdateStrategy(UpdateStrategyInterface):
+    def update(self, item):
```
Comment (vytenisgelazius) [inline python/Strategies/backstage_passes.py]: ditto return typehint

check whole diff

File: python/strategy_context.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+from handlers.aged_brie import AgedBrieUpdateStrategy
+from handlers.sulfuras import SulfurasUpdateStrategy
+from handlers.backstage_passes import BackstagePassesUpdateStrategy
+from handlers.conjured_item import ConjuredUpdateStrategy
+from handlers.normal_item import NormalItemUpdateStrategy
+
+class StrategyContext:
+    def get_strategy(self, item) -> object:
+        concert_substrings = ["concert", "backstage", "pass"]
+        if "brie" in item.name.lower():
```
Comment (vytenisgelazius) [inline python/strategy_context.py]: move this to variable & reuse it in all if/else statements

File: python/tests/unit/test_backstage_passes.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import unittest
+from gilded_rose import Item, GildedRose
+
+class BackstagePassesUpdateStrategyTest(unittest.TestCase):
```
Comment (vytenisgelazius) [inline python/tests/unit/test_backstage_passes.py:4]: use data provider here

File: python/tests/unit/test_aged_brie.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        items = [Item("Aged Brie", 5, 50)]
+        gilded_rose = GildedRose(items)
+        gilded_rose.update_quality()
+        self.assertEqual(50, items[0].quality)
+    
+    def test_brie_sell_in_should_decrease(self) -> None:
+        items = [Item("Aged Brie", 5, 10)]
+        gilded_rose = GildedRose(items)
+        gilded_rose.update_quality()
+        self.assertEqual(4, items[0].sell_in)
```
Comment (vytenisgelazius) [inline python/tests/unit/test_aged_brie.py]: try to combine these into one test, which would have one assert, but would check all the properties

File: python/tests/unit/test_conjured_item.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import unittest
+from gilded_rose import Item, GildedRose
+
+class BackstagePassesUpdateStrategyTest(unittest.TestCase):
```
Comment (vytenisgelazius) [inline python/tests/unit/test_conjured_item.py]: wut

File: python/Strategies/backstage_passes.py (hunk 2 above)
Comment (liutkauskasM12) [reply python/Strategies/backstage_passes.py]: done

File: python/Interfaces/UpdateStrategyInterface.py (hunk 1 above)
Comment (liutkauskasM12) [reply python/Interfaces/UpdateStrategyInterface.py]: Done as suggested

File: python/strategy_context.py (hunk 3 above)
Comment (liutkauskasM12) [reply python/strategy_context.py]: item_name is now used

File: python/tests/unit/test_aged_brie.py (hunk 5 above)
Comment (liutkauskasM12) [reply python/tests/unit/test_aged_brie.py]: Normal test cases were moved to assertion with an array. Edge cases stayed

File: python/tests/unit/test_conjured_item.py (hunk 6 above)
Comment (liutkauskasM12) [reply python/tests/unit/test_conjured_item.py]: fixed. Changed to ConjuredItemUpdateStrategyTest

File: python/tests/unit/test_backstage_passes.py (hunk 4 above)
Comment (liutkauskasM12) [reply python/tests/unit/test_backstage_passes.py:4]: Will do with next iteration of changes

**Review by vytenisgelazius — APPROVED** [review]
